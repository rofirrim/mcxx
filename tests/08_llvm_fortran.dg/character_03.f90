! <testinfo>
! test_generator="config/mercurium-fortran-llvm run"
! test_stdout="character_03.f90.out"
! </testinfo>
PROGRAM MAIN
    IMPLICIT NONE
    CHARACTER(LEN=10) :: A, B

    A = "ABC"
    B = "ABC"
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "ABC"
    B = "ABD"
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "ABD"
    B = "ABC"
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "ABC"
    B = "DEF"
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "DEF"
    B = "ABC"
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "ABC"
    B = "ABC  "
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "ABC"
    B = "ABD  "
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "ABC   "
    B = "ABC  D"
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "ABC   "
    B = "ABC   "
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B

    A = "ABC   "
    B = "ABC"
    PRINT *, A < B
    PRINT *, A > B
    PRINT *, A <= B
    PRINT *, A >= B
    PRINT *, A == B
    PRINT *, A /= B
END PROGRAM MAIN

