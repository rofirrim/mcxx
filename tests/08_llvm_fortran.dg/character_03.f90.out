 F
 F
 T
 T
 T
 F
 T
 F
 T
 F
 F
 T
 F
 T
 F
 T
 F
 T
 T
 F
 T
 F
 F
 T
 F
 T
 F
 T
 F
 T
 F
 F
 T
 T
 T
 F
 T
 F
 T
 F
 F
 T
 T
 F
 T
 F
 F
 T
 F
 F
 T
 T
 T
 F
 F
 F
 T
 T
 T
 F
